controllers/storagecluster/blackbox_exporter.go

```diff
@@ -0,0 +1,584 @@
+/*
+Copyright 2025 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"slices"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/defaults"
+	"github.com/red-hat-storage/ocs-operator/v4/version"
+	"go.uber.org/multierr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	blackboxExporterName   = "odf-blackbox-exporter"
+	blackboxServiceAccount = "odf-blackbox-exporter"
+	blackboxSCCName        = "odf-blackbox-scc"
+	blackboxConfigMapName  = "odf-blackbox-exporter-config"
+	portBlackboxHTTP       = "http"
+	blackboxScrapeInterval = "30s"
+	// DefaultBlackboxExporterImage is the fallback image if env var is missing
+	DefaultBlackboxExporterImage = "quay.io/prometheus/blackbox-exporter:latest"
```

Comment:
Use this flow https://github.com/red-hat-storage/ocs-operator/blob/28e6583ec44e5b78b9df3ed19258d3bb2b97e932/controllers/storagecluster/storagecluster_controller.go#L49